src/de/blau/android/imageryoffset/BackgroundAlignmentActionModeCallback.java

```diff
@@ -163,11 +163,20 @@ protected void onPreExecute() {
 	    	
 			BoundingBox bbox = Application.mainActivity.getMap().getViewBox();
 			double centerLon = (bbox.getLeft() + ((long)bbox.getRight() - (long)bbox.getLeft())/2L) / 1E7d;
+			Integer radius = params[0];
+			String radiusString = radius != null && radius > 0 ? String.valueOf(radius) : "";
+			Uri uriBuilder = Uri
+					.parse(offsetServer).buildUpon()
```

Comment:
You could do the `parse` call immediately when you retrieve the value for `offsetServer` instead of parsing several times.